circadian/models.py

```diff
@@ -1,878 +1,950 @@
+        else:
+            state = trajectory(time)
+            x = np.cos(state[2])
+            y = np.sin(state[2])
+    return np.angle(x + complex(0,1) * y)
+
+# %% ../nbs/00_models.ipynb 79
+@patch_to(Hannay19TP)
+def amplitude(self,
+              trajectory: DynamicalTrajectory=None, # trajectory to calculate the amplitude. If None, the current trajectory is used
+              time: float=None # a time point to calculate the amplitude at. If None, the amplitude is calculated for the entire trajectory
+              ) -> float:
+    if trajectory is None:
+        trajectory = self.trajectory
+    else: 
+        if not isinstance(trajectory, DynamicalTrajectory):
+            raise ValueError("trajectory must be a DynamicalTrajectory")
+    if time is None:
+        time = trajectory.time
+        amplitude = trajectory.states[:, 0]
+    else:
+        if not isinstance(time, (float, int)):
+            raise ValueError("time must be a float or an int")
+        else:
+            state = trajectory(time)
+            amplitude = state[0] 
+    return amplitude
+
+# %% ../nbs/00_models.ipynb 81
+@patch_to(Hannay19TP)
+def cbt(self,
+        trajectory: DynamicalTrajectory=None, # trajectory to calculate the cbt. If None, the current trajectory is used
+        ) -> np.ndarray:
+    "Finds the core body temperature minumum markers for the model along a trajectory as the times where the phase is pi"
+    if trajectory is None:
+        trajectory = self.trajectory
+    else:
+        if not isinstance(trajectory, DynamicalTrajectory):
+            raise ValueError("trajectory must be a DynamicalTrajectory")
+    inverted_x = -np.cos(trajectory.states[:,2])
+    cbt_min_idxs, _ = find_peaks(inverted_x)
+    return trajectory.time[cbt_min_idxs]
+
+# %% ../nbs/00_models.ipynb 83
+@patch_to(Hannay19TP)
+def dlmos(self,
+          trajectory: DynamicalTrajectory=None # trajectory to calculate the dlmo. If None, the current trajectory is used
+          ) -> np.ndarray:
+    "Finds the Dim Light Melatonin Onset (DLMO) markers for the model along a trajectory"
+    if trajectory is None:
+        trajectory = self.trajectory
+    else:
+        if not isinstance(trajectory, DynamicalTrajectory):
+            raise ValueError("trajectory must be a DynamicalTrajectory")
+    return self.cbt(trajectory) - self.cbt_to_dlmo
+
+# %% ../nbs/00_models.ipynb 87
+class Jewett99(CircadianModel):
+    "Implementation of Jewett's 1999 model from the article 'Revised Limit Cycle Oscillator Model of Human Circadian Pacemaker'"
+    def __init__(self, params=None):
+        default_params = {
+            'taux': 24.2, 'mu': 0.13, 'G': 19.875,
+            'beta': 0.013, 'k': 0.55, 'q': 1.0/3,
+            'I0': 9500, 'p': 0.6, 'alpha_0': 0.16,
+            'phi_ref': 0.8, 'cbt_to_dlmo': 7.0}
+        num_states = 3 # x, xc, n
+        num_inputs = 1 # light
+        default_initial_condition = np.array([-0.3, -1.13, 0.0])
```

Comment:
Let's define default_initial_condition to be the IC at "midnight" that you get when you loop 8 hours of darkness from midnight to midnight + 8, followed by 16 hr 1000 lux light until equilibrated, for all models

Reply:
I couldn't find a clean way of automatically defining the proper default initial condition without running into problems with self definition (can't simulate until I define the model's properties, but I need to provide a default ic to define the model's properties).

I ended up adding a new cell where I get the initial condition via `_get_default_intial_condition(model, num_loops)` and then copy/paste that output on the model definition.